test/functional/wallet_backwards_compatibility.py

```diff
@@ -178,18 +178,25 @@ def run_test(self):
         node_master_wallets_dir = node_master.wallets_path
         node_v19_wallets_dir = node_v19.wallets_path
         node_v17_wallets_dir = node_v17.wallets_path
-        node_v16_wallets_dir = node_v16.chain_path
+        node_v16_wallets_dir = node_v16.wallets_path
         node_master.unloadwallet("w1")
         node_master.unloadwallet("w2")
         node_master.unloadwallet("w3")
 
         for node in legacy_nodes:
             # Copy wallets to previous version
             for wallet in os.listdir(node_master_wallets_dir):
-                shutil.copytree(
-                    os.path.join(node_master_wallets_dir, wallet),
-                    os.path.join(self.nodes_wallet_dir(node), wallet)
-                )
+                if self.major_version_equals(node, 16):
+                    # 0.16 node expect the wallet to be in the wallet dir but as a plain file rather than in directories
+                    shutil.copyfile(
+                        node_master.wallets_path / wallet / "wallet.dat",
+                        node.wallets_path / wallet,
+                    )
+                else:
+                    shutil.copytree(
+                        node_master_wallets_dir / wallet,
+                        node.wallets_path / wallet,
+                    )
```

Comment:
In df850c25:

Could simplify this:

```suggestion
            # Copy wallets to previous version
            for wallet in os.listdir(node_master_wallets_dir):
                source = node_master_wallets_dir / wallet
                if self.major_version_equals(node, 16):
                    # 0.16 node expect the wallet to be in the wallet dir but as a plain file rather than in directories
                    source / "wallet.dat"
                shutil.copytree(source, node.wallets_path / wallet)
```

Reply:
Done